src/primitives.jl

```diff
@@ -89,42 +89,48 @@ flipped_geom = GO.apply(GI.PointTrait, geom) do p
     (GI.y(p), GI.x(p))
 end
 """
-apply(f, ::Type{Target}, geom; kw...) where Target = _apply(f, Target, geom; kw...)
+Base.@constprop :aggressive apply(f::F, ::Type{Target}, geom; threaded=false, kw...) where {F,Target} =
+    _apply(f, Target, geom; threaded=_threadtype(threaded), kw...)
+
```

Comment:
```suggestion

#=
You might note the use of this strange macro, `Base.@constprop :aggressive`.  This tells Julia that the return type of the method depends on the value of its input arguments, and allows it to specialize compilation to increase type-stability.

We are also passing the threading specification as types, not simple boolean values.  This is, again, to help compilation - with a type to hold on to, it's easier for the compiler to separate threaded and non-threaded code paths.
=#

```

Reply:
I should go through and write this stuff for everything